server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -345,9 +347,33 @@ private <C extends Collector, T> T search(Weight weight, CollectorManager<C, T>
 
     @Override
     public void search(List<LeafReaderContext> leaves, Weight weight, Collector collector) throws IOException {
-        collector.setWeight(weight);
-        for (LeafReaderContext ctx : leaves) { // search each subreader
-            searchLeaf(ctx, weight, collector);
+        try {
+            collector.setWeight(weight);
+            for (LeafReaderContext ctx : leaves) { // search each subreader
+                searchLeaf(ctx, weight, collector);
+            }
+        } catch (@SuppressWarnings("unused") TimeExceededException e) {
+            // ideally we should just throw an error here if we don't want partial results
+            timeExceeded = true;
```

Comment:
Maybe we should pass the flag `partialResults` to this index searcher so we throw an error if we do not want partial results?

Reply:
I think I would keep it like this, knowing that this is close enough to lucene's timeout handling and we will want to move to lucene support soon.